### PR TITLE
Future proof RLAYER_USE_EXPLICIT_IV by checking dtls versions directly.

### PR DESCRIPTION
**Pull request by fwh-dc**

### CODE REVIEW

File: ssl/record/methods/recmethod_local.h [hunk 1]
```diff
@@ -395,7 +395,8 @@ void ossl_rlayer_fatal(OSSL_RECORD_LAYER *rl, int al, int reason,
 
 #define RLAYER_USE_EXPLICIT_IV(rl) ((rl)->version == TLS1_1_VERSION \
                                     || (rl)->version == TLS1_2_VERSION \
-                                    || (rl)->isdtls)
+                                    || (rl)->version == DTLS1_VERSION \
+                                    || (rl)->version == DTLS1_2_VERSION)
```
Comment (mattcaswell) [inline ssl/record/methods/recmethod_local.h:399]: What about DTLS1_BAD_VER?

**Comment by mattcaswell** [comment]
There are CI failures...most likely because of the issue highlighted in my comment.

File: ssl/record/methods/recmethod_local.h (hunk 1 above)
Comment (fwh-dc) [reply ssl/record/methods/recmethod_local.h:400]: You are right!

**Review by mattcaswell — APPROVED** [review]

**Review by shahsb — APPROVED** [review]

**Comment by t8m** [comment]
> Kindly update the Description of this PR.
> 
> Also, please ensure that the commit message contains the issue number `#fixes issue_no`

I am sorry but do not understand this request. IMO the description is OK as is. And there is no issue being fixed here. It is a standalone PR.

**Review by t8m — APPROVED** [review]

**Comment by t8m** [comment]
Merged to the master branch. Thank you for your contribution.

**Pull request closed without merging**